File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const answer = [];
    const days = [];
    for(const [index, progress] of progresses.entries()){
        let num = progress
        let day = 0
        while(num + speeds[index] < 100){
            num += speeds[index]
            day++
        }
        days.push(day)
    }
    
    while(days.length !== 0){
        let count = 1
        let temp = days.shift()
        while(temp >= days[0]){
            days.shift()
            count++
        }
        answer.push(count)
    }
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    days = []
    answer = []
    for p, s in zip(progresses, speeds):
        day = 0
        total = p
        while total < 100:
            total += s
            day += 1
        days.append(day)
    cur_day = days[0]
    cnt = 1
    for day in days[1:]:
        if day <= cur_day:
            cnt += 1
        else:
            answer.append(cnt)
            cnt = 1
            cur_day = day
    answer.append(cnt)
    
    return answer
